src/EditorFeatures/CSharpTest/CodeRefactorings/UseExplicitOrImplicitType/UseExplicitTypeRefactoringTests.cs

```diff
@@ -423,6 +423,378 @@ static void Main()
+    }
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment3()
+        {
+            var code = @"
+#nullable enable
+
+class C
+{
+    private static string s_data;
+
+    static void Main()
+    {
+        var[||] v = s_data;
+        v = GetNullableString();
+    }
+
+    static string? GetNullableString() => null;
+}";
+
+            var expected = @"
+#nullable enable
+
+class C
+{
+    private static string s_data;
+
+    static void Main()
+    {
+        string? v = s_data;
+        v = GetNullableString();
+    }
+
+    static string? GetNullableString() => null;
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Foreach1()
+        {
+            var code = @"
+#nullable enable
+
+using System.Collections;
+
+class C
+{
+    static void Main()
+    {
+        foreach (var[||] item in new string?[] { """", null })
+        {
+        }
+    }
+}";
+
+            var expected = @"
+#nullable enable
+
+using System.Collections;
+
+class C
+{
+    static void Main()
+    {
+        foreach (string? item in new string?[] { """", null })
+        {
+        }
+    }
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Foreach2()
+        {
+            var code = @"
+#nullable enable
+
+using System.Collections;
+
+class C
+{
+    static void Main()
+    {
+        foreach (var[||] item in new string[] { """" })
+        {
+        }
+    }
+}";
+
+            var expected = @"
+#nullable enable
+
+using System.Collections;
+
+class C
+{
+    static void Main()
+    {
+        foreach (string item in new string[] { """" })
+        {
+        }
+    }
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Lambda1()
+        {
+            var code = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    private static string s_data = """";
+
+    static void Main()
+    {
+        Action a = () => {
+            var[||] v = s_data;
+            v = GetNullableString();
+        };
+    }
+
+    static string? GetNullableString() => null;
+}";
+
+            var expected = @"
+#nullable enable
+
+using System;
+
+class C
+{
+    private static string s_data = """";
+
+    static void Main()
+    {
+        Action a = () => {
+            string? v = s_data;
+            v = GetNullableString();
+        };
+    }
+
+    static string? GetNullableString() => null;
+}";
+
+            await TestInRegularAndScriptWhenDiagnosticNotAppliedAsync(code, expected);
+        }
+
+        [Fact]
+        public async Task TestNullabilityAssignment_Lambda2()
```

Comment:
Do we also need a test that if you're doing var that is outside a lambda, and the null assignment is _inside_ the lambda, that we see that correctly?